src/Chainweb/Pact/SPV/Hyperlane.hs

```diff
@@ -0,0 +1,276 @@
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE ViewPatterns #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Chainweb.Pact.SPV.Hyperlane where
+
+import Control.Error
+import Control.Lens hiding (index)
+import Control.Monad (when, unless)
+import Control.Monad.Catch
+import Control.Monad.Except
+
+import qualified Data.ByteString as B
+import qualified Data.ByteString.Builder as Builder
+import qualified Data.ByteString.Base16 as B16
+import qualified Data.ByteString.Short as BS
+import Data.DoubleWord
+import Data.Decimal
+import Data.Default (def)
+import Data.Foldable (foldl')
+import Data.Traversable (forM)
+import Data.Ratio
+import qualified Data.Map.Strict as M
+import Data.Text (Text)
+import qualified Data.Text as Text
+import qualified Data.Text.Encoding as Text
+import qualified Data.Vector as V
+
+import qualified Crypto.Secp256k1 as ECDSA
+
+import Ethereum.Misc hiding (Word256)
+
+import Pact.Types.Runtime
+
+import Chainweb.Pact.SPV.Hyperlane.Binary
+import Chainweb.Utils.Serialization (putRawByteString, runPutS, runGetS, putWord32be)
+
+-- | Parses the object and evaluates Hyperlane command
+evalHyperlaneCommand :: Object Name -> ExceptT Text IO (Object Name)
+evalHyperlaneCommand (_objectMap . _oObject -> om)
+  | Just (TLitString storageLocation) <- M.lookup "storageLocation" om
+  , Just (TLitString sig) <- M.lookup "signature" om
+  = recoverAddressValidatorAnnouncement storageLocation sig
+
+  | Just (TLitString message) <- M.lookup "message" om
+  , Just (TLitString metadata) <- M.lookup "metadata" om
+  , Just (TList validators _ _) <- M.lookup "validators" om
+  , Just (TLitInteger threshold) <- M.lookup "threshold" om
+  =
+    let
+      convert (TLitString v) = Just v
+      convert _ = Nothing
+    in verifySignatures message metadata (V.mapMaybe convert validators) (fromInteger threshold)
```

Comment:
I don't like the idea of throwing out validators that are not strings, I prefer an error. Something like this:

```suggestion
  | Just (TLitString message) <- M.lookup "message" om
  , Just (TLitString metadata) <- M.lookup "metadata" om
  , Just (TList (traverse preview _TLitString -> Just validators) _ _) <- M.lookup "validators" om
  , Just (TLitInteger threshold) <- M.lookup "threshold" om
  =
    in verifySignatures message metadata validators (fromInteger threshold)
```